pilot/pkg/serviceregistry/serviceentry/servicediscovery.go

```diff
@@ -677,3 +681,42 @@ func selectorChanged(old, curr model.Config) bool {
 	n := curr.Spec.(*networking.ServiceEntry)
 	return !reflect.DeepEqual(o.WorkloadSelector, n.WorkloadSelector)
 }
+
+// automatically allocates IPs for service entry services without an address field
+// if the hostname is not a wildcard, or when resolution is not NONE. The IPs are allocated
+// from the 240.240.0.0 subnet that is not reachable outside the pod. When DNS capture is
+// enabled, Envoy will resolve the DNS to these IPs. The listeners for TCP services will also
+// be set up on these IPs.
+func autoAllocateIPs(services []*model.Service) []*model.Service {
```

Comment:
method that is changing it's parameter as side-effect and returns the same thing is bad karma.

I assume there is no sync across istiod replicas - i.e. same service will have different auto-allocated IPs,
since they are local to a single envoy. Please include this in the comments in service ( if I guessed right ).

The algo is complicated enough to deserve a RFC - not clear this range is not used by anyhting in normal clusters, it is not reserved in any way. 

I would add a feature flag to disable auto-alloc - the rest of the DNS interception is great, but auto-alloc
seems to be a separate feature that is not quite well defined.

Reply:
the IP range is reserved and unused across the internet. Its the same range we use for multicluster replicated control planes for the .global services. The auto allocated IP is not used until the DNS is enabled in envoy. So its just data that sits in each service object